main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -33,303 +33,518 @@ object Safety {
     //
     // Collect all errors.
     //
-    val defErrors = root.defs.flatMap {
-      case (_, defn) => visitDef(defn, root)
-    }
 
-    val instanceErrors = visitSendable(root)
+    for {
+      _ <- Validation.foldRight(root.defs.toList)(root.toSuccess) {
```

Comment:
A few things here:

1. We never do multiline blocks inside `for`
2. Like Magnus said, we should avoid `for` when dealing with `Validation` since it stops at the first error.
3. `Validation.foldRight` does the same: stopping at the first error.
4. Yielding `errors` seems very suspicious to me. In general we yield the "good" part rather than the error.

What I would expect here is something like the following:
1. Use `Validation.traverse`  to go over the all the defs in the function. Assign the result to e.g., `defsVal`
2. Also use `Validation.traverse` to go over all the instances with `visitSendable`. Assign the result to e.g. `instsVal`
3. Map over `defsVal` and `instsVal` with `mapN`, returning a new root where we replace the `defs` and `insts` field.